core/field_variable.js

```diff
@@ -65,7 +65,8 @@ Blockly.FieldVariable = function(varName, opt_validator, opt_variableTypes,
    * variable.
    * @type {string}
    */
-  this.defaultVariableName = varName || '';
+  this.defaultVariableName =
+      typeof varName === 'string' || varName instanceof String ? varName : '';
```

Comment:
I'm surprised you need both of these conditions, or either of them really based on the types here. varName can only be null or string, so the existing code theoretically should have been fine. (or maybe varName !== null)... it looks like this line is upsetting closure though.

Reply:
Updated.